vcpkg/users/buildsystems/cmake-integration.md

```diff
@@ -6,6 +6,22 @@ ms.date: 11/30/2022
 
 # CMake Integration
 
+It is important to understand when and how vcpkg executes its dependency acquisition and build steps in the CMake context.
+As background: by default, the first time CMake configures a build, it runs internal search routines to locate a viable
+"toolchain" (compiler, linker, etc.). This search happens when the first `project()` command is encountered in `CMakeLists.txt`.
+
+In order to customize this process, CMake supports "toolchain" files, which are really just CMake scripts.
+When a toolchain is specified, CMake **evaluates the `CMAKE_TOOLCHAIN_FILE` as a script**, just like `include("${CMAKE_TOOLCHAIN_FILE}")`
+This evaluation typically results in a set of variable definitions containing the paths of necessary build tools (along with other
+build parameters, such as cross-compilation flags).
+
+But because the toolchain file is just a CMake script, it is not limited to defining toolchain-related variables. vcpkg uses this
+fact as a hook into the CMake generation process. Evaluation of the `vcpkg.cmake` toolchain file effectively runs
+`execute_process("vcpkg install")`, after other precursor steps.
+
+**Therefore: all vcpkg-provided dependencies are downloaded and compiled the first time `project()` is called**. A key consequence
+of this sequence is that all CMake-level variables impacting vcpkg itself must be defined _before_ `project()`.
```

Comment:
We should clarify that automatic dependency installation only happens in manifest mode. In classic mode, vcpkg limits to making what was already installed in the command-line available to `find_<package|library|path>` calls.

Reply:
Thanks for the review, I reworded to hopefully address this! (and hopefully make the linter happy)